sympy/physics/continuum_mechanics/column.py

```diff
@@ -0,0 +1,330 @@
+from sympy.solvers import linsolve, solve
+from sympy.core import Symbol, diff, symbols
+from sympy import dsolve, Function, Derivative, Eq, cos, sin, sqrt, tan
+from sympy.core.symbol import Dummy
+from sympy.printing import sstr
+
+class Column(object):
+    """
+    A column is a structural member designed to undertake axial
+    compressive loads. A column is characterized by its
+    cross-sectional profile(second moment of area), its length and
+    its material.
+
+    Examples
+    ========
+
+    There is a solid round bar 3 m long with second-moment I is used as a
+    column with both the ends pinned. Young's modulus of the Column is E.
+    The buckling load applied is 78KN
+
+    >>> from sympy.physics.continuum_mechanics.column import Column
+    >>> from sympy import Symbol, symbols
+    >>> E, I, P = symbols('E, I, P', positive=True)
+    >>> c = Column(3, E, I, 78000, top="pinned", bottom="pinned")
+    >>> c.end_conditions
+    {'bottom': 'pinned', 'top': 'pinned'}
+    >>> c.boundary_conditions
+    {'deflection': [(0, 0), (3, 0)], 'slope': [(0, 0)]}
+    >>> c.moment()
+    78000*y(x)
```

Comment:
What represents `y(x)` here? Shouldn't the `Function` be well defined here?

Reply:
y(x) is the deflection (in the y-direction) which is a function of x.
Should I make it defl(x) instead?

Reply:
Nah, I think `y(x)`  should be fine here. But is there a purpose of representing the moment in terms of `deflection` rather than in terms of `x`?

Reply:
But the deflection is determined only after the user calls `solve_slope_deflection()`.
So I guess for doing this we might need to create a variable to check whether deflection has been determined and then accordingly return the `moment`.
I am not sure whether to do this.